src/tests/test_utils/common_test_utils/include/common_test_utils/data_utils.hpp

```diff
@@ -190,7 +190,7 @@ void inline fill_data_random(T* pointer,
         start_from = 0;
     }
     for (std::size_t i = 0; i < size; i++) {
-        pointer[i] = static_cast<T>(start_from + static_cast<T>(random.Generate(k_range)) / k);
```

Comment:
@andrei-cv is it required? Let's revert

Reply:
random.Generate overflows fp16

Reply:
Got it, LGTM
@iefode please, take a look

Reply:
@andrei-cv @vurusovs 
Opened questions:
* What if we support FP64?
* moreover float (f32) > fp16...
* we cast values to `<T>`

Reply:
Ticket for AR: [CVS-125993](https://jira.devtools.intel.com/browse/CVS-125993)